xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/shareinline.vm

```diff
@@ -48,8 +49,12 @@
         #end ## @
         #if (!$recipients.contains($recipient) && "$!recipient" != '')
           #set ($discard = $recipients.add($recipient))
-          #set ($result = $xwiki.mailsender.sendMessageFromTemplate($sender, $recipient, ${NULL}, ${NULL}, "${xcontext.locale}", 'XWiki.SharePage', {}))
-          #if ($result != 0)
+          ## Send the email for current recipient
+          #set ($mailParameters = {'from' : $sender, 'to': $recipient, 'language' : $xcontext.language})
+          #set ($message = $services.mailsender.createMessage('template', $templateReference, $mailParameters))
```

Comment:
Missing some check for errors. See for ex:

* https://github.com/xwiki/xwiki-platform/blob/1c43781b80eaae33031d26f31ad6c29863a52467/xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/ResetPassword.xml#L132
* https://github.com/xwiki/xwiki-platform/blob/1c43781b80eaae33031d26f31ad6c29863a52467/xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/ResetPassword.xml#L139
* https://github.com/xwiki/xwiki-platform/blob/1c43781b80eaae33031d26f31ad6c29863a52467/xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/ResetPassword.xml#L143